### PR TITLE
[WIP][ClassContent] updated CategoryManager and content rest controller to accept 'only_visible' option

### PR DESCRIPTION
**Pull request by eric-chau**
With this ``only_visible`` option we can easily display into content selector only the contents that are visibles and availables as drag and droppable content.

@hbaptiste this will be useful for you, you will only need to add a query parameter into the url, example: ``GET /rest/1/classcontent?category=Article&only_online``

This PR still in progress, I will add some tests.

### CODE REVIEW

File: Rest/Controller/ClassContentController.php [hunk 1]
```diff
@@ -565,9 +570,13 @@ private function getClassContentByTypeAndUid($type, $uid, $hydrateDraft = false,
      *
      * @return null|Paginator
      */
-    private function getClassContentByCategory($name, $start, $count)
+    private function getClassContentByCategory($name, $onlyVisible, $start, $count)
```
Comment (fkroockmann) [inline Rest/Controller/ClassContentController.php]: You break the compatibility no ?

Comment (mickaelandrieu) [inline Rest/Controller/ClassContentController.php]: yes, and no.

This is a private function so this is not part of "public " api.

But, I suggest this function definition => 
``
private function getClassContentByCategory($name, $start, $count, $onlyVisible)
``

Comment (eric-chau) [inline Rest/Controller/ClassContentController.php]: This is not a BC cause it's a private method and I let this as second argument for argument order coherence.

**Comment by mickaelandrieu** [comment]
@eric-chau tell me when you consider this PR as done.

**Comment by mickaelandrieu** [comment]
ping @eric-chau let's take time to finish this one tommorrow :)

**Comment by eric-chau** [comment]
This PR was developed for @hbaptiste needs but seems that it's not the case anymore. I close this PR.

**Pull request closed without merging**